### DISCUSSION TITLE
Angular: VariantProps binds to 'any' when compoundVariants is defined

### DISCUSSION THREAD

**Discussion by Nxtivision** [joe-bell/cva#217] [Q&A]
**Describe the bug**
I use this library with Angular and TailwindCSS. It's working perfectly fine, until I use `compoundVariants`
Here is a simplified example of the problem:

```typescript
const buttonVariants = cva(
  '...all my CSS Classes',
  {
    variants: {
	variant: {
  	  flat: '',
	  outline: 'border',
	  link: 'underline-offset-4 hover:underline',
	},
	color: {
	  primary: '',
	  secondary: '',
	  destructive: '',
	},
  },
  compoundVariants: [...],
});
type ButtonVariants = VariantProps<typeof buttonVariants>;
```

And inside my component, I have something like this:

```typescript
export class UiButtonComponent implements OnChanges {
	@Input()
	variant: ButtonVariants['variant'] = 'flat';
	@Input()
	color: ButtonVariants['color'] = 'primary';

       // ...
}
```

At this moment, it works fine, no problem, types are correctly recognized. But once I go to the HTML, I lose the type (see Picture 1)

As we can see, `variant` input is recognized as `any`, which is kinda weird because all my other components work perfectly fine. The problem also appears with all the other inputs of this component. So I tried to remove the `compoundVariants` (see Picture 2)

Removing the `compoundVariants` fixed the problem. Is it a bug ? :) 

**To Reproduce**

Steps explained previously seems to be sufficient.

**Expected behavior**
In my template, I would like to have my types when I use my component

**Screenshots**
### Picture 1

![2023-09-12_15h43_13](https://github.com/joe-bell/cva/assets/89700030/2d373d63-96e2-4752-af0d-84120ada020e)

### Picture 2

![2023-09-12_15h44_56](https://github.com/joe-bell/cva/assets/89700030/ca4b4d86-c97d-4399-9e1b-7b5677a68b56)

**Comment by joe-bell** [comment]
Sorry! I’m not an expert on Angular to provide any advice here, but I’ll move this to a discussion and hopefully someone else can chip in 🙏🏼

**Comment by adrekic** [comment]
This is still an issue, it has nothing to do with the js/ts framework. Reproducing in a typescript project.
